libs/editor/WordPressEditor/src/main/java/org/wordpress/android/editor/AztecEditorFragment.java

```diff
@@ -883,17 +882,31 @@ public void onClick(DialogInterface dialog, int id) {
 
                 builder.setNegativeButton(getString(R.string.cancel), new DialogInterface.OnClickListener() {
                     public void onClick(DialogInterface dialog, int id) {
+                        try {
+                            // only retry if id is valid
+                            Integer.parseInt(localMediaId);
+                            mEditorFragmentListener.onMediaRetryClicked(localMediaId);
+                        } catch (NumberFormatException e) {
+                            // invalid id
+                        }
                         dialog.dismiss();
                     }
                 });
 
+                // stop the media upload while the dialog is being shown
+                // otherwise the upload might finish and we won't find the element for removal
```

Comment:
This seems pretty drastic. Even canceling the dialog will cause a re-upload of the image, consuming power and network traffic, both quite important resources on the mobile.

The commit title mentions that the upload will "resume" but I don't think it actually resumes. I think it restarts from the beginning, right?

> we won't find the element for removal

I don't think this is be a big problem. The removal will fail silently and the user can delete the image again if they really want to. Silently is not optimal so, we can put up a Toast message if we want, saying that the image has finished uploading anyway.